client/src/resources/cert-and-project-map.ts

```diff
@@ -714,10 +729,14 @@ certMap.forEach(cert => {
   // URLs aren't rendered on the settings page
   if (cert.title !== 'Legacy Full Stack') {
     if (cert.title.startsWith('Legacy')) {
+      // eslint-disable-next-line
+      // @ts-ignore
       legacyProjectMap[cert.title] = cert.projects;
```

Comment:
1

Reply:
So, what you can do is this.  Instead of defining the `certMapTypes` type, you can just add 'as const' to the end of the array.

```ts
] as const;
```
on line 755

that lets you extract the title types themselves

```
const titles = certMap.map(({ title }) => title);
type Title = typeof titles[number];
```
and use them to type your `legacyProjectMap` and `projectMap`s via

```ts
const legacyProjectMap: Partial<Record<Title, unknown>> = {};
const projectMap: Partial<Record<Title, unknown>> = {};
```
We need `Partial` here because otherwise `Record` will complain about the missing titles.

Reply:
Thanks! I've sent you a PM via chat about some questions I had about this.